modules/apps/digital-signature/digital-signature-web/test/js/components/document-previewer/DocumentPreviewerWrapper.js

```diff
@@ -17,6 +17,11 @@ import React from 'react';
 
 import DocumentPreviewerWrapper from '../../../../src/main/resources/META-INF/resources/js/components/document-previewer/DocumentPreviewerWrapper';
 
+jest.mock('frontend-js-web', () => ({
+	...jest.requireActual('frontend-js-web'),
+	sub: jest.fn((string, arg) => string.replace(/x/g, arg)),
```

Comment:
How come we are mocking `sub` here? I would think that we should expect sub to work as needed.

Reply:
This new mock basically makes sure that all `x`s will be replaced with the arg, as that's how the test is written, so the mock supports that => `document-x-of-x` should be `document-1-of-1`, not `document-1,1-of-x`

Reply:
That seems a little irrelevant to this test and the rest of this code though. Won't this just be another line of code to maintain without much purpose? Maybe I am missing something here, but what I'm trying to understand is how adding this mock improves our reliability on these tests, wouldn't we rather test against the real implementation of `sub`?

Reply:
You're right, the mock was super specific, so I replaced it (in https://github.com/liferay-frontend/liferay-portal/pull/2701/commits/d686d054f5e0d0856c1051829c77de44f2f0edba) with a mock that we already use across the codebase (but for `Liferay.Util.sub`).
The reason why there are some new mocks in some of these tests is because those tests relied on the globally available `Liferay.Util.utilName` mock defined in [liferay-frontend-projects](https://github.com/kresimir-coko/liferay-frontend-projects/blob/master/projects/npm-tools/packages/npm-scripts/src/jest/mocks/Liferay.js)

Reply:
If we totally remove the mocks to  `sub` that comes from frontend-js-web, what ends up being the issue? Is it just snapshots that need to be updated? I think ideally we wouldn't want to mock it at all because it is in a sense the "true" implementation of it. Whereas, for the global mock we had previously was used because we previously didn't have access to the sub function on the global namespace